party_time/party_time.lua

```diff
@@ -313,6 +389,62 @@ ui.display(function()
         end
     end
 
+    -- Dialog for Selecting new Party|Alliance Leader
+    if unhandled_leader then
+        local height = 36
+        for index, member in pairs(party) do
+            if type(index) == "number" then
+                if index <= 6 and member.id ~= player.id then
+                    height = height + 24
+                elseif party.alliance.alliance_leader_id == player.id and (member.id == party.alliance.party_2_leader_id or member.id == party.alliance.party_3_leader_id) then
+                    height = height + 24
+                end
+            end
+            height = height + 6
+        end
+        leader_dialog.state.height = height
+
+        leader_dialog.state, leader_dialog.closed = ui.window('leader_dialog', leader_dialog.state, function()
+            ui.location(11, 5)
+            ui.text('    Select your new\nparty | alliance leader')
+
+            --loop members
+            local button_y = 50
+            for index, member in pairs(party) do
+                if type(index) == "number" then
+                    if index <= 6 and member.id ~= player.id then
+                        ui.location(23,button_y)
+                        --TODO: Once button text is fixed for custom sizes use -> ui.size(90,25)
+                        if ui.button(member.name, member.name) then
+                            command.input('/pcmd leader ' .. member.name)
+                            unhandled_leader = false
+                        end
+                        button_y = button_y + 24
+                    elseif party.alliance.alliance_leader_id == player.id and (member.id == party.alliance.party_2_leader_id or member.id == party.alliance.party_3_leader_id) then
+                        ui.location(23,button_y)
+                        --TODO: Once button text is fixed for custom sizes use -> ui.size(90,25)
+                        if ui.button(member.name,  member.name) then
+                            command.input('/acmd leader ' .. member.name)
+                            unhandled_leader = false
+                        end
+                        button_y = button_y + 24
```

Comment:
This entire `elseif` block is the same as the one before except for the `/acmd`. And the conditions are the same as above. I would write functions to check whether the current member is in the party or alliance. Then you could do something like:
```lua
local in_party = check_party(member)
local in_alliance = check_alliance(member)
if in_party or in_alliance then
    local prefix = in_party and '/pcmd' or  '/acmd'
    -- stuff
        command.input(prefix .. member.name)
    -- more stuff
end
```